test/cpp/qps/client.h

```diff
@@ -280,19 +274,21 @@ class Client {
         if (entry.used()) {
           histogram_.Add(entry.value());
         }
+        bool done = false;
         if (!thread_still_ok) {
           gpr_log(GPR_ERROR, "Finishing client thread due to RPC error");
-          done_ = true;
+          done = true;
         }
-        if (done_) {
+        done = done || (gpr_atm_acq_load(&client_->thread_pool_done_) !=
+                        static_cast<gpr_atm>(0));
+        if (done) {
```

Comment:
Right, I kept it around only because I thought that it hurt readability otherwise, but now that I think about it, a logical `||` operator outside of an `if` statement is a less frequent construct and should be avoided. I'll make the change that you suggest.